MastodonSDK/Sources/MastodonUI/Scene/ComposeContent/Attachment/AttachmentViewModel+Compress.swift

```diff
@@ -92,3 +93,57 @@ extension AttachmentViewModel {
         }
     }   // end func
 }
+
+extension AttachmentViewModel {
+    @AttachmentViewModelActor
+    func compressImage(data: Data, sizeLimit: SizeLimit) throws -> Output {
+        let maxPayloadSizeInBytes = sizeLimit.image ?? 10 * 1024 * 1024
+
+        guard let image = KFCrossPlatformImage(data: data)?.kf.normalized,
+              var imageData = image.kf.pngRepresentation()
+        else {
+            throw AttachmentError.invalidAttachmentType
+        }
+        
+        repeat {
+            guard let image = KFCrossPlatformImage(data: imageData) else {
+                throw AttachmentError.invalidAttachmentType
+            }
+
+            if imageData.kf.imageFormat == .PNG {
+                // A. png image
+                if imageData.count > maxPayloadSizeInBytes {
+                    guard let compressedJpegData = image.jpegData(compressionQuality: 0.8) else {
```

Comment:
Do you think it would make sense to try other formats like WebP or AVIF?

Reply:
That's possible if the webp format is included in `Mastodon.Entity.Instance.Configuration.MediaAttachments.supportedMIMETypes`.